crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -112,9 +114,15 @@ impl<'a> Resolver<'a> {
         }
 
         for unused_var in unused_vars.iter() {
-            self.push_err(ResolverError::UnusedVariable {
-                ident_id: *unused_var,
-            });
+            if !self
+                .interner
+                .ident_name(unused_var)
+                .starts_with(IDENT_ERROR_PREFIX)
```

Comment:
So this will skip over the $error identifiers that were created in the parser? If so, I'm wonder if there is a way to make this more apparent that IDENT_ERROR_PREFIX is from the parser's hardcoded value and or whether we need to use an identifier in the parser for $error?

Reply:
Yes, its to avoid reporting errors twice (and avoid showing variables named "$error" to the user).
It may be clearer here to have a ERROR_IDENT instead and share that between the parser and name resolver

Reply:
For the apparent rationale: for example, creating the constant in the parser and then using it in the parser and here. 

Although I wonder why we cannot have Token::Error?

Reply:
What I was suggesting was to emit a Token::Error on all parser errors or I guess ExpressionKind::Error, however I do not believe this to be a good idea as it would require things like let statements to change from `ident : Ident` to `ident : ExpressionKind` for no substantial reason